account_wallet/models/account_wallet_type.py

```diff
@@ -0,0 +1,94 @@
+# © 2015  Laetitia Gangloff, Acsone SA/NV (http://www.acsone.eu)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class AccountWalletType(models.Model):
+    _name = "account.wallet.type"
+    _description = "Wallet Type"
+    _check_company_auto = True
+
+    name = fields.Char(translate=True, required=True)
+    sequence_id = fields.Many2one(
+        comodel_name="ir.sequence",
+        string="Wallet Sequence",
+        copy=False,
+        check_company=True,
+        help="This field contains the information related to the numbering "
+        "of the wallet of this type.",
+        required=True,
+    )
+    account_id = fields.Many2one(
+        comodel_name="account.account",
+        string="Account",
+        ondelete="restrict",
+        index=True,
+        required=True,
+    )
+    journal_id = fields.Many2one(
+        comodel_name="account.journal",
+        string="Journal",
+        ondelete="restrict",
+        help="Journal use to empty the wallet",
+        required=True,
+    )
+    product_id = fields.Many2one(
+        comodel_name="product.product",
+        string="Product",
+        ondelete="restrict",
+        help="Product use to fill the wallet",
+    )
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        string="Company",
+        default=lambda self: self.env.company,
+        required=True,
+    )
+
+    # TODO: Check if this is necessary and if model cannot be simplified
+    # @api.constrains('product_id', 'account_id', 'journal_id')
+    # def _check_account(self):
```

Comment:
by digging a little into this module, I think that : this contrains (commented) could be removed.

check on journal_id seems based on obsolete code. journal doesn't seems to be configured in that release of wallet module. (or maybe I missed something).

In fact, bad configuration is always possible if we : 
- create a correct ``wallet.type``
- then go on the product form, and change the account configuration. 

As the final income account used for a product is defined by many fields. (computed by the function ``product_id._get_product_accounts()["income"]``) 

However, https://github.com/acsone/wallet/pull/2  could reduce misconfiguration.

Reply:
Done here : https://github.com/acsone/wallet/pull/4